modules/hrm/includes/class-employee.php

```diff
@@ -1682,12 +1683,45 @@ public function update_employment_status( $args = [] ) {
 
         $args['date'] = erp_current_datetime()->modify( $args['date'] )->format( 'Y-m-d H:i:s' );
 
+        /**
+         * We need to check if any history exists in the future date.
+         * If exists, we will consider it as an old history and so it
+         * will not affect the executive values of the employee.
+         */
         $future_history = $this->erp_user->histories()
                                         ->where( 'module', $args['module'] )
-                                        ->where( 'date', '>', $args['date'] )
-                                        ->count();
+                                        ->where( 'date', '>', $args['date'] );
+
+        if ( ! empty( $args['id'] ) ) {
+
+            /**
+             * If we've reached at this block, that means
+             * we're editing a history. So we need to make sure
+             * the validation process, while making comparison,
+             * excludes the history which is being edited.
+             */
+            $future_history = $future_history->where( 'id', '!=', $args['id'] );
+
+            /**
+             * As we can only edit the current history,
+             * we need to make sure the given date is
+             * eligibe for the current history.
+             * If any old history exists after this date,
+             * it cannot be used for the current history
+             * and so we will abandon the process.
+             */
+            if ( (int) $future_history->count() > 0 ) {
```

Comment:
can `count()` ever give us non-integer number? 
If it always return an integer, here parsing to `int` is not necessary.